src/narrowdown.h

```diff
@@ -120,8 +120,7 @@ class NarrowDown
       if (!pred(entries.back(), pseudoKey)) {
         std::stringstream ss;
         ss << "Dirent table is not properly sorted:\n";
-        ss << "  #0: " << key[0] << "/" << key.substr(1) << "\n";
-        ss << "  #1: " << nextKey[0] << "/" << nextKey.substr(1);
+        ss << "PseudoKey " << pseudoKey << " should be after (or equal) previously generated " << pred.getKeyContent(entries.back()) << "\n";
```

Comment:
Now that the format of this message is not the same as in `FileImpl::checkDirentOrder()` maybe it's better to leave it as an `ASSERT()`?

Reply:
No. The more I think about this the more I'm convinced we should rethink how we should handle errors.

- ASSERT should be for coding error (ours).
- ZimFileFormatError should be invalid data (zim file)
- Specific errors (should be for invalid arguments (coming from the user code)

This is not always what is done but we should slowly (at least) go this way.

Here, if the pseudoKey is not after the previous entry, it is because the zim file is wrong. So we should raise a ZimFileFormatError